dot/services_integration_test.go

```diff
@@ -0,0 +1,343 @@
+
+	rpcSrvc, err := createRPCService(cfg, ns, stateSrvc, coreSrvc, networkSrvc, nil, sysSrvc, nil)
+	require.NoError(t, err)
+	require.NotNil(t, rpcSrvc)
+}
+
+func TestCreateBABEService(t *testing.T) {
+	cfg := NewTestConfig(t)
+	require.NotNil(t, cfg)
+
+	genFile := NewTestGenesisRawFile(t, cfg)
+	require.NotNil(t, genFile)
+
+	cfg.Core.Roles = types.FullNodeRole
+	cfg.Init.Genesis = genFile
+
+	err := InitNode(cfg)
+	require.NoError(t, err)
+
+	stateSrvc, err := createStateService(cfg)
+	require.NoError(t, err)
+
+	ks := keystore.NewGlobalKeystore()
+	kr, err := keystore.NewSr25519Keyring()
+	require.NoError(t, err)
+	ks.Babe.Insert(kr.Alice())
+
+	ns, err := createRuntimeStorage(stateSrvc)
+	require.NoError(t, err)
+	err = loadRuntime(cfg, ns, stateSrvc, ks, &network.Service{})
+	require.NoError(t, err)
+
+	dh, err := createDigestHandler(cfg.Log.DigestLvl, stateSrvc)
+	require.NoError(t, err)
+
+	coreSrvc, err := createCoreService(cfg, ks, stateSrvc, &network.Service{}, dh)
+	require.NoError(t, err)
+
+	bs, err := createBABEService(cfg, stateSrvc, ks.Babe, coreSrvc, nil)
+	require.NoError(t, err)
+	require.NotNil(t, bs)
+}
+
+func TestCreateGrandpaService(t *testing.T) {
+	cfg := NewTestConfig(t)
+	require.NotNil(t, cfg)
+
+	genFile := NewTestGenesisRawFile(t, cfg)
+	require.NotNil(t, genFile)
+
+	cfg.Core.Roles = types.AuthorityRole
+	cfg.Init.Genesis = genFile
+
+	err := InitNode(cfg)
+	require.NoError(t, err)
+
+	stateSrvc, err := createStateService(cfg)
+	require.NoError(t, err)
+
+	ks := keystore.NewGlobalKeystore()
+	kr, err := keystore.NewEd25519Keyring()
+	require.NoError(t, err)
+	ks.Gran.Insert(kr.Alice())
+
+	ns, err := createRuntimeStorage(stateSrvc)
+	require.NoError(t, err)
+
+	err = loadRuntime(cfg, ns, stateSrvc, ks, &network.Service{})
+	require.NoError(t, err)
+
+	dh, err := createDigestHandler(cfg.Log.DigestLvl, stateSrvc)
+	require.NoError(t, err)
+
+	gs, err := createGRANDPAService(cfg, stateSrvc, dh, ks.Gran, &network.Service{}, nil)
+	require.NoError(t, err)
+	require.NotNil(t, gs)
+}
+
+var addr = flag.String("addr", "localhost:8546", "http service address")
+var testCalls = []struct {
+	call     []byte
+	expected []byte
+}{
+	{[]byte(`{"jsonrpc":"2.0","method":"system_name","params":[],"id":1}`),
+		[]byte(`{"id":1,"jsonrpc":"2.0","result":"gossamer"}` + "\n")}, // working request
+	{[]byte(`{"jsonrpc":"2.0","method":"unknown","params":[],"id":2}`),
+		[]byte(`{"error":{"code":-32000,"data":null,"message":"rpc error method unknown not found"},"id":2,
+"jsonrpc":"2.0"}` + "\n")}, // unknown method
+	{[]byte{},
+		[]byte(`{"jsonrpc":"2.0","error":{"code":-32600,"message":"Invalid request"},"id":0}` + "\n")}, // empty request
+	{[]byte(`{"jsonrpc":"2.0","method":"chain_subscribeNewHeads","params":[],"id":3}`),
+		[]byte(`{"jsonrpc":"2.0","result":1,"id":3}` + "\n")},
+	{[]byte(`{"jsonrpc":"2.0","method":"state_subscribeStorage","params":[],"id":4}`),
+		[]byte(`{"jsonrpc":"2.0","result":2,"id":4}` + "\n")},
+}
```

Comment:
Move these inlined in `TestNewWebSocketServer`

Reply:
moved.